src/Engine/Component/Trigger.cpp

```diff
@@ -0,0 +1,31 @@
+#include "Trigger.hpp"
+#include "../Trigger/SuperTrigger.hpp"
+
+using namespace Component;
+
+Trigger::Trigger() {
+    // ADD CONSTRUCTOR
+}
+
+Trigger::~Trigger() {
+
+    if(m_trigger != nullptr)
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') keyword 'if' not followed by a single space [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=vera%2B%2B%3AT008)